chalk-ir/src/lib.rs

```diff
@@ -929,8 +930,16 @@ impl<I: Interner> Const<I> {
             ConstValue::InferenceVar(_) => false,
             ConstValue::Placeholder(_) => false,
             ConstValue::Concrete(_) => false,
+            // We assume for now that an unevaluated const expression doesn't contain bound vars
+            ConstValue::Unevaluated(_) => false,
```

Comment:
Hmm. Not sure that this assumption is correct. If the `Unevaluated` had an expression like `{N + 1}`, then there `N` would probably be a bound var and we would want to shift. But I haven't thought about this enough to know either way right now.

Reply:
But I guess for now, this is probably fine. But I think I would like a followup issue filed with or without a `FIXME` here.

Reply:
Which is why I said "for now".